src/containers/edit-profile/password-security-tab/PasswordSecurityTab.tsx

```diff
@@ -1,34 +1,136 @@
 import AppButton from '~/components/app-button/AppButton'
 import AppTextField from '~/components/app-text-field/AppTextField'
 import ConfirmDialog from '~/components/confirm-dialog/ConfirmDialog'
 import TitleWithDescription from '~/components/title-with-description/TitleWithDescription'
+import { snackbarVariants } from '~/constants'
 
 import useForm from '~/hooks/use-form'
+import useAxios from '~/hooks/use-axios'
+import { useAppDispatch } from '~/hooks/use-redux'
+import useInputVisibility from '~/hooks/use-input-visibility'
+
+import { AuthService } from '~/services/auth-service'
+import { openAlert } from '~/redux/features/snackbarSlice'
 
 import { styles } from '~/containers/edit-profile/password-security-tab/PasswordSecurityTab.styles'
-import { confirmPassword, password } from '~/utils/validations/login'
-import { ButtonVariantEnum, InputEnum, SizeEnum, FormValues } from '~/types'
 
-const PasswordSecurityTab = () => {
+import {
+  ButtonVariantEnum,
+  ComponentEnum,
+  ButtonTypeEnum,
+  InputEnum,
+  SizeEnum,
+  FormValues,
+  ErrorResponse,
+  EditProfileTabUserProps
+} from '~/types'
+import {
+  initialValues,
+  validations
+} from '~/containers/edit-profile/password-security-tab/PasswordSecurityTab.constants'
+
+const PasswordSecurityTab: FC<EditProfileTabUserProps> = ({ user }) => {
   const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+
   const [isConfirmOpen, setIsConfirmOpen] = useState(false)
 
-  const { data, handleInputChange, errors, handleBlur, resetData } =
-    useForm<FormValues>({
-      initialValues: {
-        currentPassword: '',
-        password: '',
-        confirmPassword: ''
-      },
-      validations: { password, confirmPassword }
-    })
+  const handleResponse = () => {
+    dispatch(
+      openAlert({
+        severity: snackbarVariants.success,
+        message: 'editProfilePage.profile.successMessage'
+      })
+    )
+  }
+
+  const changePassword = useCallback(
+    (data: { password: string; currentPassword: string }) => {
+      return AuthService.changePassword(user._id, data)
+    },
+    [user._id]
+  )
+
+  const handleResponseError = (error?: ErrorResponse) => {
+    if (error?.code === 'INCORRECT_CREDENTIALS') {
+      handleErrors('password', t('common.errorMessages.samePasswords'))
+    } else {
+      handleErrors('currentPassword', t('common.errorMessages.currentPassword'))
+    }
+  }
+
+  const { loading, fetchData: sendChangedPassword } = useAxios({
+    service: changePassword,
+    onResponse: handleResponse,
+    onResponseError: handleResponseError,
+    fetchOnMount: false
+  })
+
+  const {
+    data,
+    handleSubmit,
+    handleInputChange,
+    errors,
+    handleBlur,
+    resetData,
+    resetErrors,
+    handleErrors
+  } = useForm<FormValues>({
+    onSubmit: async () => {
+      resetErrors()
+      await sendChangedPassword({
+        password: data.password,
+        currentPassword: data.currentPassword
+      })
+    },
```

Comment:
extract into separate function and pass it here

Reply:
Done! Looks more readeble)